src/Moq/Protected/ProtectedMock.cs

```diff
@@ -219,9 +221,10 @@ public void VerifySet<TProperty>(string propertyName, Times times, object value)
 			ThrowIfPublicSetter(property, typeof(T).Name);
 			Guard.CanWrite(property);
 
+			var expression = GetSetterExpression(property, ItExpr.IsAny<TProperty>());
 			// TODO should consider property indexers
 			// TODO should receive the parameter here
```

Comment:
Check whether these TODOs. I am suspecting that indexers work now (let's add tests to find out for sure!), and I cannot make sense of the second TODO.

Reply:
The TODOs are still valid:

* The first one is relevant because validation before the `GetSetterExpression` call might fail for indexers.

* The second one could mean that it should be possible for `VerifySet` to receive the right-hand side of the assignment invocation to be verified, instead of just assuming `It.IsAny<T>()`.